diff --git a/Makefile b/Makefile
index fa8cb76..e9116b4 100644
--- a/Makefile
+++ b/Makefile
@@ -612,7 +612,7 @@ quiet_cmd_busybox__ ?= LINK    $@
       "$(core-y)" \
       "$(libs-y)" \
       "$(LDLIBS)" \
-      "$(CONFIG_EXTRA_LDLIBS)" \
+      "$(subst \",,$(CONFIG_EXTRA_LDLIBS))" \
       && $(srctree)/scripts/generate_BUFSIZ.sh --post include/common_bufsiz.h
 
 # Generate System.map
diff --git a/Makefile.flags b/Makefile.flags
index 1cec5ba..303dfae 100644
--- a/Makefile.flags
+++ b/Makefile.flags
@@ -146,7 +146,7 @@ export SYSROOT=$(CONFIG_SYSROOT)
 endif
 
 # libm may be needed for dc, awk, ntpd
-LDLIBS += m
+# LDLIBS += m
 # Android has no separate crypt library
 # gcc-4.2.1 fails if we try to feed C source on stdin:
 #  echo 'int main(void){return 0;}' | $(CC) $(CFLAGS) -lcrypt -o /dev/null -xc -
diff --git a/coreutils/cat.c b/coreutils/cat.c
index 4b34149..e7d2ac2 100644
--- a/coreutils/cat.c
+++ b/coreutils/cat.c
@@ -27,7 +27,7 @@
 //config:	help
 //config:	Display nonprinting characters as escape sequences
 
-//applet:IF_CAT(APPLET(cat, BB_DIR_BIN, BB_SUID_DROP))
+//applet:IF_CAT(APPLET_NOFORK(cat, cat, BB_DIR_BIN, BB_SUID_DROP, cat))
 
 //kbuild:lib-$(CONFIG_CAT) += cat.o
 
 diff --git a/coreutils/ls.c b/coreutils/ls.c
 index b69b804..6b6dd0a 100644
 --- a/coreutils/ls.c
 +++ b/coreutils/ls.c
@@ -92,7 +92,7 @@
 //config:	configurable, and the output may not be legible on
 //config:	many output screens.
 
-//applet:IF_LS(APPLET_NOEXEC(ls, ls, BB_DIR_BIN, BB_SUID_DROP, ls))
+//applet:IF_LS(APPLET_NOFORK(ls, ls, BB_DIR_BIN, BB_SUID_DROP, ls))
 
 //kbuild:lib-$(CONFIG_LS) += ls.o
+
+diff --git a/editors/vi.c b/editors/vi.c
+index 0000000..0000000 100644
+--- a/editors/vi.c
++++ b/editors/vi.c
+@@ -176,7 +176,7 @@
+ //config:	Enable more verbose reporting of the results of yank, change,
+ //config:	delete, undo and substitution commands.
+ 
+-//applet:IF_VI(APPLET(vi, BB_DIR_BIN, BB_SUID_DROP))
++//applet:IF_VI(APPLET_NOFORK(vi, vi, BB_DIR_BIN, BB_SUID_DROP, vi))
+ 
+ //kbuild:lib-$(CONFIG_VI) += vi.o
 
diff --git a/scripts/trylink b/scripts/trylink
index 2255dee..8b846e7 100755
--- a/scripts/trylink
+++ b/scripts/trylink
@@ -84,7 +84,8 @@ A_FILES="$6"
 # a real utmp library in LDLIBS, dropping it "works" but resulting binary
 # does not work properly).
 LDLIBS="$7"
-CONFIG_EXTRA_LDLIBS="$8"
+shift 7
+CONFIG_EXTRA_LDLIBS="$*"
 
 # The --sort-section option is not supported by older versions of ld
 SORT_SECTION="-Wl,--sort-section,alignment"
diff --git a/coreutils/cp.c b/coreutils/cp.c
index ee40af5..e13b7f5 100644
--- a/coreutils/cp.c
+++ b/coreutils/cp.c
@@ -30,7 +30,7 @@
 //config:	default y
 //config:	depends on FEATURE_CP_LONG_OPTIONS
 
-//applet:IF_CP(APPLET_NOEXEC(cp, cp, BB_DIR_BIN, BB_SUID_DROP, cp))
+//applet:IF_CP(APPLET_NOFORK(cp, cp, BB_DIR_BIN, BB_SUID_DROP, cp))
 /* NOEXEC despite cases when it can be a "runner" (cp -r LARGE_DIR NEW_DIR) */
 
 //kbuild:lib-$(CONFIG_CP) += cp.o
diff --git a/coreutils/ln.c b/coreutils/ln.c
index 2dcf79c..e762adf 100644
--- a/coreutils/ln.c
+++ b/coreutils/ln.c
@@ -12,7 +12,7 @@
 //config:	help
 //config:	ln is used to create hard or soft links between files.
 
-//applet:IF_LN(APPLET_NOEXEC(ln, ln, BB_DIR_BIN, BB_SUID_DROP, ln))
+//applet:IF_LN(APPLET_NOFORK(ln, ln, BB_DIR_BIN, BB_SUID_DROP, ln))
 
 //kbuild:lib-$(CONFIG_LN) += ln.o
 
diff --git a/coreutils/mv.c b/coreutils/mv.c
index cf6169a..0783ee1 100644
--- a/coreutils/mv.c
+++ b/coreutils/mv.c
@@ -17,7 +17,7 @@
 //config:	help
 //config:	mv is used to move or rename files or directories.
 
-//applet:IF_MV(APPLET_NOEXEC(mv, mv, BB_DIR_BIN, BB_SUID_DROP, mv))
+//applet:IF_MV(APPLET_NOFORK(mv, mv, BB_DIR_BIN, BB_SUID_DROP, mv))
 /* NOEXEC despite cases when it can be a "runner" (mv LARGE_DIR OTHER_FS) */
 
 //kbuild:lib-$(CONFIG_MV) += mv.o
diff --git a/coreutils/rm.c b/coreutils/rm.c
index d000129..4c834b1 100644
--- a/coreutils/rm.c
+++ b/coreutils/rm.c
@@ -16,7 +16,7 @@
 //config:	help
 //config:	rm is used to remove files or directories.
 
-//applet:IF_RM(APPLET_NOEXEC(rm, rm, BB_DIR_BIN, BB_SUID_DROP, rm))
+//applet:IF_RM(APPLET_NOFORK(rm, rm, BB_DIR_BIN, BB_SUID_DROP, rm))
 /* was NOFORK, but then "rm -i FILE" can't be ^C'ed if run by hush */
 
 //kbuild:lib-$(CONFIG_RM) += rm.o
diff --git a/coreutils/stat.c b/coreutils/stat.c
index 2c2909e..1966633 100644
--- a/coreutils/stat.c
+++ b/coreutils/stat.c
@@ -35,7 +35,7 @@
 //config:	Without this, stat will not support the '-f' option to display
 //config:	information about filesystem status.
 
-//applet:IF_STAT(APPLET_NOEXEC(stat, stat, BB_DIR_BIN, BB_SUID_DROP, stat))
+//applet:IF_STAT(APPLET_NOFORK(stat, stat, BB_DIR_BIN, BB_SUID_DROP, stat))
 
 //kbuild:lib-$(CONFIG_STAT) += stat.o
 
